src/audio/mixin_mixout.c

```diff
@@ -1020,6 +1111,73 @@ static int mixout_get_attribute(struct comp_dev *dev, uint32_t type, void *value
 	return 0;
 }
 
+static int mixin_set_large_config(struct comp_dev *dev, uint32_t param_id, bool first_block,
+				  bool last_block, uint32_t data_offset_or_size, char *data)
+{
+	struct ipc4_mixer_mode_config *cfg;
+	struct mixin_data *mixin_data;
+	int i;
+	uint32_t sink_index;
+	uint16_t g;
+
+	if (param_id != IPC4_MIXER_MODE) {
+		comp_err(dev, "mixin_set_large_config() unsupported param_id: %u", param_id);
+		return -EINVAL;
+	}
+
+	if (!(first_block && last_block)) {
+		comp_err(dev, "mixin_set_large_config() data is expected to be sent as one chunk");
+		return -EINVAL;
+	}
+
+	/* for a single chunk data, data_offset_or_size is size */
+	if (data_offset_or_size < sizeof(struct ipc4_mixer_mode_config)) {
+		comp_err(dev, "mixin_set_large_config() too small data size: %u",
+			 data_offset_or_size);
+		return -EINVAL;
+	}
+
+	assert(data_offset_or_size <= SOF_IPC_MSG_MAX_SIZE);
```

Comment:
return an error too, I think

Reply:
That's a check for data_offset_or_size should be less than IPC mailbox memory window size. In theory, it should never fail :) . In the worst case, if it fails, nothing terrible happen: data is casted to struct and used within boundaries of the struct. The worst thing is that dcache_invalidate_region() might be called with too large size.
So I'd rather remove that assert() at all, than replacing it with error report.